src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -628,11 +636,13 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                 ImmutableArray<PendingBranch> returns = walker.Analyze(ref badRegion, initialLocalState);
                 snapshotBuilderOpt?.ExitWalker(walker.SaveSharedState(), previousSlot);
                 diagnostics?.AddRange(walker.Diagnostics);
+                variableTypes = walker._variableTypes.ToImmutableDictionary();
```

Comment:
>variableTypes = walker._variableTypes.ToImmutableDictionary(); [](start = 16, length = 62)

Should be moved to the caller so the caller can avoid the allocation if not needed.

Reply:
This can't be done. Freeing the walker frees the variable types PooledDictionary, and I don't think it's correct to move responsibility for freeing this dictionary out of the main walker itself.

---
In reply to: [305068984](https://github.com/dotnet/roslyn/pull/37322#discussion_r305068984) [](ancestors = 305068984)

Reply:
This is a `private` method. The only callers are in the walker.

---
In reply to: [305452556](https://github.com/dotnet/roslyn/pull/37322#discussion_r305452556) [](ancestors = 305452556,305068984)